File: Java/LinkedLists/SwapPairs.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
  /// <summary>
  /// Given the head of a singly linked list swap the order of each pair of
  /// elements.
  /// </summary>
  /// <param name="head">The head of the list.</param>
  /// <remarks>
  /// Example:
  /// Input: (1) -> (2) -> (3) -> (4) -> (5)
  /// Output: (2) -> (1) -> (4) -> (3) -> (5)
  /// </remarks>
  public Node swapPairs(Node head) {
    return head;
  }

  public static void main(String args[]) {
    Solution solution = new Solution();

    System.out.println("\nTesting odd input...");
    solution.testOddInput();

    System.out.println("\nTesting even input...");
    solution.testEvenInput();

    System.out.println("\nTesting small input...");
    solution.testSmallInput();

    System.out.println("\nTesting null input...");
    solution.testNullInput();
  }

  public void testOddInput() {
    Node head = createList(5);
    System.out.println(String.format("Input: %s", head.toString()));

    head = swapPairs(head);
    System.out.println(String.format("Output: %s", head.toString()));
  }

  public void testEvenInput() {
    Node head = createList(6);
    System.out.println(String.format("Input: %s", head.toString()));

    head = swapPairs(head);
    System.out.println(String.format("Output: %s", head.toString()));
  }

  public void testSmallInput() {
    Node head = new Node(1);
    System.out.println(String.format("Input: %s", head.toString()));

    head = swapPairs(head);
    System.out.println(String.format("Output: %s", head.toString()));
  }

  public void testNullInput() {
    System.out.println("Input: null");

    Node head = swapPairs(null);
    System.out.println(String.format("Output: %s", head == null ? "null" : head.toString()));
  }

  public Node createList(int count) {
    List<Node> nodes = new ArrayList<Node>();

    for (int i = 1; i <= count; i++) {
      nodes.add(new Node(i));
    }

    for (int i = 0; i < nodes.size() - 1; i++) {
      nodes.get(i).setNext(nodes.get(i + 1));
    }

    return nodes.get(0);
  }

  public class Node {
    private Node next;

    private int data;

    public Node(int data) {
      this.data = data;
    }

    public Node getNext() {
      return this.next;
    }

    public void setNext(Node next) {
      this.next = next;
    }

    public int getData() {
      return this.data;
    }

    public String toString() {
      if (next != null) {
        return String.format("(%s) -> %s", data, next);
      } else {
        return String.format("(%s) -> null", data);
      }
    }
  }
}

File: DotNet/LinkedLists/SwapPairs.cs
using System;
using System.Linq;

namespace CodeChallenges.DotNet.LinkedLists
{
    public class Solution
    {
        // <summary>
        //     Given the head of a singly linked list,
        //     swap the order of each pair of elements.
        // </summary>
        // <param name="head">
        //     The head of the list.
        // </param>
        // <remarks>
        //     class Node {
        //        int data;
        //        Node next;
        //     }
        //
        //     Example:
        //         Input:  null
        //         Output: null
        //
        //         Input:  (1)
        //         Output: (1) -> null
        //
        //         Input:  (1) -> (2) -> (3) -> (4) -> (5)
        //         Output: (2) -> (1) -> (4) -> (3) -> (5) -> null
        //
        //         Input:  (1) -> (2) -> (3) -> (4) -> (5) -> (6)
        //         Output: (2) -> (1) -> (4) -> (3) -> (6) -> (5) -> null
        // </remarks>

        public Node SwapPairs(Node head)
        {
            return head;
        }

        public class Node
        {
            public Node Next { get; set; }

            public int Data { get; }

            public Node(int data)
            {
              this.Data = data;
            }

            public override string ToString()
            {
                if (Next != null)
                {
                    return $"({Data}) -> {Next}";
                }
                else
                {
                    return $"({Data}) -> null";
                }
            }
        }

        static void Main(string[] args)
        {
            Solution solution = new Solution();

            Console.WriteLine("\nTesting odd input...");
            solution.TestOddInput();

            Console.WriteLine("\nTesting even input...");
            solution.TestEvenInput();

            Console.WriteLine("\nTesting small input...");
            solution.TestSmallInput();

            Console.WriteLine("\nTesting null input...");
            solution.TestNullInput();
        }

        public void TestOddInput()
        {
            Node head = CreateList(5);
            Console.WriteLine($"Input:  {head.ToString()}");

            head = SwapPairs(head);
            Console.WriteLine($"Output: {head.ToString()}");
        }

        public void TestEvenInput()
        {
            var head = CreateList(6);
            Console.WriteLine($"Input:  {head.ToString()}");

            head = SwapPairs(head);
            Console.WriteLine($"Output: {head.ToString()}");
        }

        public void TestSmallInput()
        {
            var head = new Node(1);
            Console.WriteLine($"Input:  {head.ToString()}");

            head = SwapPairs(head);
            Console.WriteLine($"Output: {head.ToString()}");
        }

        public void TestNullInput()
        {
            Console.WriteLine("Input:  null");

            var head = SwapPairs(null);
            Console.WriteLine(
                head == null ? "Output: null" : $"Output: {head.ToString()}"
            );
        }

        public Node CreateList(int count)
        {
            var nodes = Enumerable.Range(1, count).Select(i => new Node(i)).ToList();

            for (int i = 0; i < nodes.Count - 1; i++)
            {
                nodes[i].Next = nodes[i + 1];
            }

            return nodes[0];
        }
    }
}
